includes/wikia/helpers/FileNamespaceSanitizeHelper.php

```diff
@@ -109,4 +105,44 @@ public function stripFilesFromWikitext( $wikitext, $lang ) {
 
 		return $wikitext;
 	}
+
+	/**
+	 * @desc for a given wikitext, return an array of images or files occurences,
+	 * without brackets and without any params
+	 *
+	 * @param string $wikitext to find images or files in
+	 * @param $lang
+	 * @return array of images ['File:sefes', 'File:blabla']
+	 * or false if no images found
+	 */
+	public function getCleanFileMarkersFromWikitext( $wikitext, $lang ) {
+		$filePrefixRegex = substr( $this->getFilePrefixRegex( $lang ), 1 );
+		preg_match_all( '/\[\[' . $filePrefixRegex .'.*\]\]/U', $wikitext, $images );
+		$cleanFileMarkers = false;
+
+		if ( count( $images[0] ) ) {
+			$cleanFileMarkers = array_map( function ( $imageMarker ) {
```

Comment:
I think this block will be unneeded if you use regex like 
'/\[\[(' . $filePrefixRegex .'[^|]).*\]\]/U'
(to be verified, I think it should work)